domain/src/main/java/org/oppia/android/domain/spotlight/SpotlightStateController.kt

```diff
@@ -137,19 +136,14 @@ class SpotlightStateController @Inject constructor(
   private fun retrieveCacheStore(
     profileId: ProfileId
   ): PersistentCacheStore<SpotlightStateDatabase> {
-    val cacheStore = if (profileId in cacheStoreMap) {
-      cacheStoreMap[profileId]!!
-    } else {
-      val cacheStore = cacheStoreMap.getOrPut(profileId) {
-        cacheStoreFactory.createPerProfile(
-          CACHE_NAME,
-          SpotlightStateDatabase.getDefaultInstance(),
-          profileId
-        )
-      }
-      cacheStoreMap[profileId] = cacheStore
-      cacheStore
+    val cacheStore = cacheStoreMap.getOrPut(profileId) {
+      cacheStoreFactory.createPerProfile(
+        CACHE_NAME,
+        SpotlightStateDatabase.getDefaultInstance(),
+        profileId
+      )
     }
+    cacheStoreMap[profileId] = cacheStore
```

Comment:
I don't think you need this line, either, since ``getOrPut`` ensures that it's in the map.

Reply:
done